consensus/istanbul/core/roundstate_persist.go

```diff
@@ -0,0 +1,351 @@
+	if err != nil {
+		return err
+	}
+
+	return rsp.storeRoundState()
+}
+
+// DesiredRound implements RoundState.DesiredRound
+func (rsp *roundStatePersistence) DesiredRound() *big.Int { return rsp.delegate.DesiredRound() }
+
+// State implements RoundState.State
+func (rsp *roundStatePersistence) State() State { return rsp.delegate.State() }
+
+// Proposer implements RoundState.Proposer
+func (rsp *roundStatePersistence) Proposer() istanbul.Validator { return rsp.delegate.Proposer() }
+
+// Subject implements RoundState.Subject
+func (rsp *roundStatePersistence) Subject() *istanbul.Subject { return rsp.delegate.Subject() }
+
+// Proposal implements RoundState.Proposal
+func (rsp *roundStatePersistence) Proposal() istanbul.Proposal { return rsp.delegate.Proposal() }
+
+// Round implements RoundState.Round
+func (rsp *roundStatePersistence) Round() *big.Int { return rsp.delegate.Round() }
+
+// Commits implements RoundState.Commits
+func (rsp *roundStatePersistence) Commits() MessageSet { return rsp.delegate.Commits() }
+
+// Prepares implements RoundState.Prepares
+func (rsp *roundStatePersistence) Prepares() MessageSet { return rsp.delegate.Prepares() }
+
+// ParentCommits implements RoundState.ParentCommits
+func (rsp *roundStatePersistence) ParentCommits() MessageSet { return rsp.delegate.ParentCommits() }
+
+// Sequence implements RoundState.Sequence
+func (rsp *roundStatePersistence) Sequence() *big.Int { return rsp.delegate.Sequence() }
+
+// View implements RoundState.View
+func (rsp *roundStatePersistence) View() *istanbul.View { return rsp.delegate.View() }
+
+// GetPrepareOrCommitSize implements RoundState.GetPrepareOrCommitSize
+func (rsp *roundStatePersistence) GetPrepareOrCommitSize() int {
+	return rsp.delegate.GetPrepareOrCommitSize()
+}
+
+// GetValidatorByAddress implements RoundState.GetValidatorByAddress
+func (rsp *roundStatePersistence) GetValidatorByAddress(address common.Address) istanbul.Validator {
+	return rsp.delegate.GetValidatorByAddress(address)
+}
+
+// ValidatorSet implements RoundState.ValidatorSet
+func (rsp *roundStatePersistence) ValidatorSet() istanbul.ValidatorSet {
+	return rsp.delegate.ValidatorSet()
+}
+
+// IsProposer implements RoundState.IsProposer
+func (rsp *roundStatePersistence) IsProposer(address common.Address) bool {
+	return rsp.delegate.IsProposer(address)
+}
+
+// Preprepare implements RoundState.Preprepare
+func (rsp *roundStatePersistence) Preprepare() *istanbul.Preprepare {
+	return rsp.delegate.Preprepare()
+}
+
+// PendingRequest implements RoundState.PendingRequest
+func (rsp *roundStatePersistence) PendingRequest() *istanbul.Request {
+	return rsp.delegate.PendingRequest()
+}
+
+// PreparedCertificate implements RoundState.PreparedCertificate
+func (rsp *roundStatePersistence) PreparedCertificate() istanbul.PreparedCertificate {
+	return rsp.delegate.PreparedCertificate()
+}
+
+func (rsp *roundStatePersistence) storeRoundState() error {
```

Comment:
I think that adding a comment stating that the reason for storing at every view be helpful (e.g. saving evidence to show that a validator did a "valid" double signing).